src/main/java/graphql/execution/ValuesResolver.java

```diff
@@ -166,7 +170,7 @@ private Object coerceValue(VariableDefinition variableDefinition, GraphQLType gr
             if (value instanceof Map) {
                 return coerceValueForInputObjectType(variableDefinition, (GraphQLInputObjectType) graphQLType, (Map<String, Object>) value);
             } else {
-                throw new CoercingParseValueException("Variables for GraphQLInputObjectType must be an instance of a Map according to the graphql specification.  The offending object was a " + value.getClass().getName());
+                throw new CoercingParseValueException("Expected type 'Map'.");
```

Comment:
Can we please put back the extended error message here.  We have had some trouble with this in the past (whereby people think they can pass any Java object but the spec says you cant...)

Say

     "Variables for input objects must be an instance of a Map according to the graphql specification."

Reply:
I worry that this error is too "internally" focused. While this may feel like a special case, isn't this just the same as above? If we could say `"Expected type 'Map' but received a 'String'."` is that good enough?

Reply:
I would prefer to have a very explicit error message: This is not a expected behaviour in any way and we should provide as much help as we can to find the error as quickly as possible.

Reply:
if there are really use cases where this error message it to 'internally' we will tackle this problem on another level (more outside).